clang/lib/CodeGen/CGExprCXX.cpp

```diff
@@ -1073,7 +1075,7 @@ void CodeGenFunction::EmitNewArrayInitializer(
       return;
     }
 
-    InitListElements = ILE->getNumInits();
+    InitListElements = ILE ? ILE->getNumInits() : CPLIE->getInitExprs().size();
```

Comment:
This is not great, but I think it will do for now.
Maybe we want to have both CXXParenListInitExpr and  InitListExpr inherit from some abstract base class that implement some of the shared functionality with CRTP. We'd avoid all these if.

I don't think we need to do that in this patch though

Reply:
```suggestion
    ArrayRef<const Expr *> InitExprs = ILE ? ILE->inits() : CPLIE->getInitExprs();
    InitListElements = InitExprs.size();
```

If you move this up from below it can look cleaner